lib/creatubbles/base_object.rb

```diff
@@ -9,8 +9,27 @@ def self.type_name
     @type_name
   end
 
+  def self.attributes
+    @attributes || []
+  end
+
+  def attributes
+    self.class.attributes
+  end
+
+  def relationships
+    self.class.relationships
+  end
+
+  def self.relationships
+    @relationships || []
+  end
+
   def self.define_attributes(atts)
-    atts.each do |att|
+    return if @attributes
```

Comment:
my main point was that in case it's called multiple times, like in:

`define_attributes :id, :title`
`define_attributes :description, :url`

it should just do the correct thing. I.e. attributes should then return a list of 4 elements. BTW, in case an attribute is added twice, it should probably just ignore that (or throw an exception)